web/modules/custom/joinup_licence/src/Controller/LicenceComparerController.php

```diff
@@ -264,13 +290,68 @@ protected function buildRow(string $category, string $label, array $items): arra
       ];
     }
 
-    $this->padWithEmptyCells($row, [
+    $this->fillContentRows($row);
+
+    return $row;
+  }
+
+  /**
+   * Fills in the header rows with an add licence option and empty cells.
+   *
+   * @param array $row
+   *   A row array.
+   */
+  protected function fillHeaderRows(array &$row): void {
+    $amount = LicenceComparerHelper::MAX_LICENCE_COUNT - count($this->licences);
+    if ($amount === 0) {
+      return;
+    }
+
+    $classes = ['licence-comparer__header', 'licence-comparer__empty'];
+    // \Drupal\Core\Render\Element\Select::processSelect does not seem
+    // to be called if the element is not rendered through a form
+    // builder. Thus, the name of the field and the default values are
+    // not set. Manually set the additional properties.
+    $row[] = [
+      'data' => [
+        'licence_search' => [
```

Comment:
We should add a hidden label for this select in order to help test find the select.